src/transformers/models/dallemega/modeling_dallemega.py

```diff
@@ -0,0 +1,1557 @@
+                This is useful if you want more control over how to convert `input_ids` indices into associated vectors
+                than the model's internal embedding lookup matrix.
+            output_attentions (`bool`, *optional*):
+                Whether or not to return the attentions tensors of all attention layers. See `attentions` under
+                returned tensors for more detail.
+            output_hidden_states (`bool`, *optional*):
+                Whether or not to return the hidden states of all layers. See `hidden_states` under returned tensors
+                for more detail.
+            return_dict (`bool`, *optional*):
+                Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+        """
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+        superconditioning_scale = superconditioning_scale if superconditioning_scale is not None else self.config.superconditioning_scale
+
+        do_superconditioning = superconditioning_scale > 1 and not self.training
+
+        # retrieve input_ids and inputs_embeds
+        if input_ids is not None and inputs_embeds is not None:
+            raise ValueError("You cannot specify both input_ids and inputs_embeds at the same time")
+        elif input_ids is not None:
+            input_shape = input_ids.size()
+            input_ids = input_ids.view(-1, input_shape[-1])
+        elif inputs_embeds is not None:
+            input_shape = inputs_embeds.size()[:-1]
+        else:
+            raise ValueError("You have to specify either input_ids or inputs_embeds")
+
+        if inputs_embeds is None:
+            inputs_embeds = self.embed_tokens(input_ids)
+        
+        if do_superconditioning:
+            input_ids_uncond = torch.ones(input_shape, dtype=torch.long, device=inputs_embeds.device) * self.config.pad_token_id
+            attention_mask_uncond = torch.zeros(input_shape, dtype=torch.long, device=inputs_embeds.device)
+            
+            inputs_embeds_unconditional = self.embed_tokens(input_ids_uncond)
+            
+            # concatenate the embeddings of the conditioned and unconditioned inputs
+            inputs_embeds = torch.cat([inputs_embeds, inputs_embeds_unconditional], dim=0)
+
+            # concatenate the attention masks of the conditioned and unconditioned inputs
+            # if attention_mask is None, create an all-ones mask
+            if attention_mask is None:
+                attention_mask = torch.ones(input_shape, dtype=torch.long, device=inputs_embeds.device)
+            attention_mask = torch.cat([attention_mask, attention_mask_uncond], dim=0)
```

Comment:
here we extend the `inputs_embeds` with `inputs_embeds_unconditional` to get the unconditional hidden states.

Reply:
Good for me!